metricflow/plan_conversion/dataflow_to_sql.py

```diff
@@ -1456,12 +1456,12 @@ def visit_semi_additive_join_node(self, node: SemiAdditiveJoinNode) -> SqlDataSe
                     ),
                 ),
             ),
-            column_alias=time_dimension_column_name,
+            column_alias=f"{time_dimension_column_name}__{node.agg_by_function.value}",
         )
         column_equality_descriptions.append(
             ColumnEqualityDescription(
                 left_column_alias=time_dimension_column_name,
-                right_column_alias=time_dimension_column_name,
+                right_column_alias=time_dimension_select_column.column_alias,
```

Comment:
yea, we don't actually access this anywhere outside the subquery. I made this change since I actually had issues with BQ where the subquery was something like this 
```sql
SELECT 
  MIN(ds) AS ds
FROM data_set
GROUP BY ds
```
this threw a `cannot group by an aggregate` error. So by making this local alias, it allowed it to group by properly. So unless we might have a use case of `MIN(ds)` anywhere outside of the subquery I assumed it would be fine to do this.

Reply:
One of the invariants is that all column aliases are generated through the column association resolver. Is this something you can handle there?

Reply:
as discussed offline, we are adding `Optional[AggregationState]` to `resolve_time_dimension_spec` and just in this semi_additive_join scenario, I’ll be passing in `AggregationState.COMPLETE` and we would render that column alias as `{time_dim}__complete` and that suffix `*__complete` may get translated to something else that’s simpler in the future.